src/txn/txn.c

```diff
@@ -1947,6 +1962,10 @@ __wt_txn_commit(WT_SESSION_IMPL *session, const char *cfg[])
     return (0);
 
 err:
+
+    /* Leave the commit generation if there are any errors. */
+    __wt_session_gen_leave(session, WT_GEN_TXN_COMMIT);
```

Comment:
Is it possible that gen_leave will be called twice from this code path? i.e: if it jumps to error after the prior gen_leave?

Reply:
Yes, there could be a situation where `__wt_session_gen_leave` will be called twice only in the error case, but I do not see any issue in calling twice. Sorry, I might be missing something.